django/db/models/expressions.py

```diff
@@ -1620,6 +1620,20 @@ def get_source_expressions(self):
     def set_source_expressions(self, exprs):
         self.condition, self.result = exprs
 
+    def resolve_expression(
+        self, query=None, allow_joins=True, reuse=None, summarize=False, for_save=False
+    ):
+        """
+        Ensure condition is always resolved with for_save=False since it's used
+        as a filter, not as an update value.
+        """
+        c = super().resolve_expression(query, allow_joins, reuse, summarize, for_save)
+        if c.condition is not None:
```

Comment:
This is only necessary when `for_save` as in other case `c` is properly resolved.

```python
if for_save and c.condition is not None
```